### PR TITLE
Fixes for array params.

### LINKED ISSUE #31: [api-client] array params methods should use append instead of set
here is an example swagger parameter definition: 
```
{
  "type": "array",
  "items": {
    "type": "string"
  },
  "name": "filters",
  "in": "query"
}
```

and here is what gets generated for that parameter

```
  listThings(filters: string[], options?: HttpOptions): Observable<models.V2RuleList> {
    const path = `/api/v2/things`;
    options = {...this.options, ...options};

    if (filters) {
      options.params = options.params.set('filters', String(filters));
    }
    return this.sendRequest<models.V2ThingList>('GET', path, options);
  }
```

The filters array ends up being joined with commas. 

__What should happen:__

The generated code for the param should look more like this: 

```
    if (filters) {
      filters.forEach(v => {
        options.params = options.params.append('filters', v);
      });
    }
```

This is unfortunately arguable since [apparently](https://stackoverflow.com/questions/24059773/correct-way-to-pass-multiple-values-for-same-parameter-name-in-get-request) there is not a clear standard. However multiple params with the same name is the most and reliable approach AFAIK so it should probably be used.

The functionality could be hidden behind a flag for backwards compatibility i.e. it uses `set` by default but `append` if `--append-array-params` is supplied during generation.

### PR DESCRIPTION
**Pull request by warmans**
closes #31 (and #36?)

So as it turns out the code already had some support for array parameters it was just not working (`isArray` was not being set). So this MR just fixes that bug and changes set to append since set in the original code doesn't really make sense.

Also in cases where there are no external definitions/models (as with the example swagger file I added) then it now suppresses the import statement since it breaks the client.

~~Finally I've also just changed the paths in the package.json scripts to use the vendored binaries instead of global version since I guess that was the intention of vendoring them in the first place (unless npm is smart about checking for a local bin before checking for a global version?).~~ yes it is.

One wierd thing is in fixing the IsArray property it seems to now generate any[] arrays instead of proper types. I'm not sure why... it'll still work but is a bit of a regression.

### CODE REVIEW

**Comment by warmans** [comment]
I don't know what's going on with the commits here. The only commit for this change is the last one. I guess the older ones will be ignored.

**Comment by vmasek** [comment]
Thank you for a PR, I started some work around this at the weekend because the problem layed much deeper (as you also mentioned). I have some fixes that are correcting the parameter types and might have conflicts with this.

I will push it and after that we'll solve the merge.

File: templates/ngx-module-export.mustache [hunk 1]
```diff
@@ -4,7 +4,10 @@ import { NgModule, ModuleWithProviders } from '@angular/core';
 import { HttpHeaders, HttpParams } from '@angular/common/http';
 import { APIClient, USE_DOMAIN, USE_HTTP_OPTIONS } from './api-client.service';
 
+{{#importDefinitions}}
```
Comment (vmasek) [inline templates/ngx-module-export.mustache]: `definitions.length` can be used for this purpose
```
{{#definitions.length}}
export * from './models';
{{/definitions.length}}
```

**Comment by warmans** [comment]
I've removed the new example I added since gcloud firestore already has some array parameters to test with.

**Comment by vmasek** [comment]
Almost all PR changes got covered by the #36 fix, however, take a closer look at the append in generated params/headers.
I have looked at doc and headers support array as param, but in params, I've used a reduce. In my test app it had worked, so let me know if there will be any further problems.

**Comment by warmans** [comment]
Cool yeah looks like it. When do you think you'll tag a new release?

**Comment by vmasek** [comment]
Tomorrow it should be out, just need to finish #8 that will change passing parameters from flat to args object (it will allow to take advantage of optional parameters)

**Comment by vmasek** [comment]
@warmans I have a troubles publishing the package, I've written an email to an npm support. After that, it should publish a `3.0.x` version to npm

**Pull request merged**